pandas/core/generic.py

```diff
@@ -1948,6 +1948,129 @@ def tail(self, n=5):
             return self
         return self.iloc[-n:]
 
+  
+    def sample(self, n=None, frac=None, replace=False, weights=None, random_state=None, axis = 0):
+        """
+        Returns a random sample of rows from object. 
+        
+        Parameters
+        ----------
+        n : int, optional
+            Number of rows to return. Cannot be used with `frac`.
+            Default = 1 if `frac` = None. 
+        frac : float, optional
+            Share of rows to return. Cannot be used with `n`. 
+        replace : boolean, optional
+            Sample with or without replacement. Default = False. 
+        weights : str or ndarray-like, optional
+            Default 'None' results in equal probability weighting. 
+            If called on a DataFrame or Panel, will also accept the name of a 
+            column as a string. Must be same length as index. 
+            If weights do not sum to 1, they will be normalized to sum to 1. 
+            Missing values in the weights column will be treated as zero. 
+            inf and -inf values not allowed. 
+        random_state : int or numpy.random.RandomState, optional
+            Seed for the random number generator (if int), or numpy RandomState 
+            object.
+        axis : int, optional
+            Axis to sample. Default = 0.
+            
+        Returns
+        -------
+        Same type as caller. 
+        """
+        
+        ###        
+        # Processing axis argument
+        ###
+        
+        # This checks validity of axis argument.
+        if axis > (self.ndim - 1):
+            raise ValueError("Axis argument out of range")
+    
+        # Store length of relevant axis of object. 
+        axis_length = self.shape[axis]    
+    
+        ###
+        # Processing random_state argument
+        ###
+    
+        # random_state may be either a RandomState object, 
+        # integer, or None. Otherwise raise an error. 
+     
+        if isinstance(random_state, int):
```

Comment:
Can you factor this out into a separate helper function? Could put it in `pandas.core.common`